contracts/HubPool.sol

```diff
@@ -80,41 +87,50 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
     event WhitelistRoute(address originToken, uint256 destinationChainId, address destinationToken);
 
     event InitiateRefundRequested(
-        uint64 indexed relayerRefundId,
         uint64 requestExpirationTimestamp,
-        uint64 poolRebalanceLeafCount,
+        uint64 unclaimedPoolRebalanceLeafCount,
         uint256[] bundleEvaluationBlockNumbers,
         bytes32 poolRebalanceRoot,
         bytes32 destinationDistributionRoot,
         address indexed proposer
     );
     event RelayerRefundExecuted(uint256 relayerRefundId, MerkleLib.PoolRebalance poolRebalance, address caller);
 
-    event RelayerRefundDisputed(uint256 relayerRefundId, address disputer);
+    event RelayerRefundDisputed(
+        address disputer,
+        SkinnyOptimisticOracleInterface.Request ooPriceRequest,
+        RefundRequest refundRequest
+    );
 
     constructor(
         uint256 _bondAmount,
         uint64 _refundProposalLiveness,
-        address _bondToken,
-        address _l1Weth,
-        address _timerAddress
-    ) Testable(_timerAddress) {
+        FinderInterface _finder,
+        bytes32 _identifier,
+        IERC20 _bondToken,
+        WETH9Like _l1Weth,
+        address _timer
+    ) Testable(_timer) {
         bondAmount = _bondAmount;
         refundProposalLiveness = _refundProposalLiveness;
-        bondToken = IERC20(_bondToken);
-        l1Weth = WETH9Like(_l1Weth);
+        bondToken = _bondToken;
+        l1Weth = _l1Weth;
+        finder = _finder;
+        identifier = _identifier;
     }
 
     /*************************************************
      *                ADMIN FUNCTIONS                *
      *************************************************/
 
     function setBondToken(address newBondToken) public onlyOwner {
+        require(refundRequest.unclaimedPoolRebalanceLeafCount == 0, "Cant set during active request");
         bondToken = IERC20(newBondToken);
         emit BondTokenSet(newBondToken);
     }
 
-    function setBondTokenFinalFeeMultiplier(uint64 newBondAmount) public onlyOwner {
+    function setBondAmount(uint64 newBondAmount) public onlyOwner {
+        require(refundRequest.unclaimedPoolRebalanceLeafCount == 0, "Cant set during active request");
```

Comment:
might want to make this a modifier if you re-use it

Reply:
also, is there any reason not to combine `setBondToken` and `setBondAmount` into `setBond(address token, uint256 amount)`?

Reply:
The only argument I could see against having this as a modifier would be that we can have more descriptive revert messages. I think we can find a more common one that covers both though. I'll revert for that.

Regarding the `setBond` that's a good point. no reason to not have just one method seeing both are restricted by the `onlyOwner`. I'll change this as well.